docs/book/src/tasks/experimental-features/cluster-classes.md

```diff
@@ -0,0 +1,455 @@
+# Cluster Class
+
+The ClusterClass feature introduces a new way to create clusters which reduces boilerplate and enables flexible and powerful customization of CAPI-created clusters.
+ClusterClass is implemented on top of existing CAPI interfaces and offers a set of tools and operations to streamline cluster lifecycle management while maintaining the same underlying API.
+
+This tutorial covers using Cluster API and Cluster Class to create a Kubernetes cluster and how to perform a one-touch Kubernetes update procedure using ClusterClass features.
+
+<aside class="note warning">
+
+<h1>Note</h1>
+
+This guide covers creating a ClusterClass cluster with the CAPD provider.
+</aside>
+
+## Installation
+
+### Common Prerequisites
+This guide is based on having the same pre-requisites as the CAPI [quick start guide].
+That is:
+- Install and setup [kubectl]  and [clusterctl]
+- Install [Kind] and [Docker]
+
+ClusterClass is currently behind a feature gate that needs to be enabled.
```

Comment:
Add this as a pre-requisite?

Reply:
This is all under pre-requisites, but I'll break it up into tool installation and then the commands.